### PR TITLE
interfaces: add display-control interface

### PR DESCRIPTION
**Pull request by jdstrand**
interfaces: add display-control interface
    
    References:
    https://forum.snapcraft.io/t/interface-to-change-display-brightness/8890

### CODE REVIEW

**Comment by jdstrand** [comment]
I only had access to two devices with different backlights and created as general a rule as I could for them. I figure we'll keep adding rules as hardware is found. Looking into this more, I have a better way.

**Comment by jdstrand** [comment]
@mvo5 and @zyga - I looked into how other software is doing this and found that they are looking at the symlinks in /sys/class/backlight, so I came up with something that will work well for primary screens (see the commit message). I also noticed some DBus services that could be in the interface and added them.

File: interfaces/builtin/common_test.go [hunk 1]
```diff
@@ -84,6 +85,15 @@ func MockEvalSymlinks(test *testutil.BaseTest, fn func(string) (string, error))
 	})
 }
 
+// MockReadDir replaces the io/ioutil.ReadDir function used inside the caps package.
+func MockReadDir(test *testutil.BaseTest, fn func(string) ([]os.FileInfo, error)) {
```
Comment (zyga) [inline interfaces/builtin/common_test.go]: Typically we return a function that restores the mocking operation. I don't oppose this but we can change that on a drive-by next time.

File: interfaces/builtin/display_control.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    peer=(label=unconfined),
+
+/sys/class/backlight/ r,
+`
+
+type displayControlInterface struct {
+	commonInterface
+}
+
+func (iface *displayControlInterface) dereferencedBacklightPaths() []string {
```
Comment (zyga) [inline interfaces/builtin/display_control.go]: This looks good.

File: interfaces/builtin/display_control.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+	}
+	return paths
+}
+
+func (iface *displayControlInterface) AppArmorConnectedPlug(spec *apparmor.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
+	// add the static rules
+	spec.AddSnippet(displayControlConnectedPlugAppArmor)
+
+	// add the detected rules
+	for _, p := range iface.dereferencedBacklightPaths() {
```
Comment (zyga) [inline interfaces/builtin/display_control.go]: I think this is still some time away but this feels like a good candidate for hot plug transformation.

Comment (jdstrand) [reply interfaces/builtin/display_control.go]: Yes, I agree (and mentioned that in the git commit ;)

File: interfaces/builtin/common_test.go (hunk 1 above)
Comment (jdstrand) [reply interfaces/builtin/common_test.go]: I was just following what we did for MockEvalSymlinks, directly above.
